src/vsr/replica.zig

```diff
@@ -8089,8 +8118,8 @@ pub fn ReplicaType(
         }
 
         fn primary_send_start_view_after_view_change(self: *Self) void {
-            assert(self.status == .normal);
-            assert(self.primary());
+            assert(self.status == .normal or self.status == .view_change);
+            assert(self.replica == self.primary_index(self.view));
```

Comment:
This is first comment I was talking about in https://github.com/tigerbeetle/tigerbeetle/pull/2416#issuecomment-2435684626! We can't assert `primary_journal_repaired` as one of the prepares *might* be corrupted after the DVC headers were first converted into SV headers. In https://github.com/tigerbeetle/tigerbeetle/commit/3095ebe3f4019eb4b4e2fbdb88acce10fbb10497, instead of adding an assert, I added a check for this:

https://github.com/tigerbeetle/tigerbeetle/blob/3095ebe3f4019eb4b4e2fbdb88acce10fbb10497/src/vsr/replica.zig#L8196-L8202

Actually, we don't need all prepares to be present before sending SV. We only need the _headers_ to be intact from op_repair_min → self.op. I had decided to not separate out `primary_journal_repaired` into `primary_journal_headers_repaired` and `primary_journal_prepares_repaired` for the sake of simplicity, since this is the only place where the distinction is required. 🤔

Reply:
I took a stab at separating them out in https://github.com/tigerbeetle/tigerbeetle/pull/2416/commits/df6caa5fb234c94f47702aab8af8e01a088ddd37, so we can refine the condition for sending SVs a bit more such that we send SVs as soon as the _header chain_ is repaired.

There is *some* code duplication across `primary_journal_repaired `, `primary_journal_headers_repaired ` and `primary_journal_prepares_repaired ` (specifically the asserts)... Thoughts?

Reply:
Ah, thanks for the clarifications! Sorry, I overlooked that comment originally. :facepalm:

Reply:
No problem dj! It turned out for the best as it made me rethink this logic and refine the conditions under which we send SV. 😄